src/common/modals/AddInstance/InstanceDuplicateName.js

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable */
+import { useDispatch, useSelector } from 'react-redux';
+import React, { useEffect, useState } from 'react';
+import { Button, Input } from 'antd';
+import fse from 'fs-extra';
+import path from 'path';
+import { transparentize } from 'polished';
+import { useInterval } from 'rooks';
+import makeDir from 'make-dir';
+import { omit } from 'lodash';
+import Modal from '../../components/Modal';
+import { closeModal } from '../../reducers/modals/actions';
+import { _getInstancesPath } from '../../utils/selectors';
+import { updateInstanceConfig } from '../../reducers/actions';
+
+const InstanceDuplicateName = ({ instanceName }) => {
+  const dispatch = useDispatch();
+  const [loading, setLoading] = useState(false);
+  const oldInstanceName = instanceName;
+  const instancesPath = useSelector(_getInstancesPath);
+  const [newInstanceName, setNewInstanceName] = useState(
+    `${oldInstanceName} copy`
+  );
+  const [alreadyExists, setAlreadyExists] = useState(false);
+  const [invalidName, setInvalidName] = useState(true);
+
+  useEffect(() => {
+    if (newInstanceName) {
+      const regex = /^[\sa-zA-Z0-9_.-]+$/;
+      const finalWhiteSpace = /[^\s]$/;
+      if (
+        !regex.test(newInstanceName) ||
+        !finalWhiteSpace.test(newInstanceName) ||
+        newInstanceName.length >= 45
+      ) {
+        setInvalidName(true);
+        setAlreadyExists(false);
+        return;
+      }
+      fse.pathExists(path.join(instancesPath, newInstanceName)).then(exists => {
+        setAlreadyExists(exists);
+        setInvalidName(false);
+      });
+    }
+  }, [newInstanceName]);
+
+  const { start, stop } = useInterval(() => {
+    if (!loading) {
+      stop();
+      dispatch(closeModal());
+    }
+  }, 200);
+
+  const duplicateInstance = async () => {
+    setLoading(true);
+    start();
+    await makeDir(path.join(instancesPath, newInstanceName));
+
+    // Copy the old config file, so the launcher can detect the instance
+    fse.copySync(
+      path.join(instancesPath, oldInstanceName, 'config.json'),
+      path.join(instancesPath, newInstanceName, 'config.json')
+    );
+
+    // Copy the old instance to the new instance folder
+    await fse.copy(
+      path.join(instancesPath, oldInstanceName),
+      path.join(instancesPath, newInstanceName)
+    );
+
+    // Remove curseforge manifest if one exists
+    if (
+      fse.pathExistsSync(
+        path.join(instancesPath, newInstanceName, 'manifest.json')
+      )
+    ) {
+      fse.rmSync(path.join(instancesPath, newInstanceName, 'manifest.json'));
+    }
+
+    // Reset the time played back to 0 and reset icon
+    dispatch(
```

Comment:
This is actually a good idea to reset time played and last played, I would not reset the icon though